src/platform/unix/mod.rs

```diff
@@ -472,7 +472,12 @@ impl OsIpcReceiverSet {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
             Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Err(ref e) if e.kind() != ErrorKind::Interrupted => {
+                return Ok(selection_results);
```

Comment:
Well, the canonical way of handling `EINTR` is using a retry loop.

In most cases it's *probably* not strictly necessary to handle it here, since typical users calling `select()` in a loop and iterating over the result each time will generally just work -- not sure though we should rely on this, as we never return an empty result otherwise as far as I can tell. On a more conceptual level, it seems less ideal to pass control back to the caller unnecessarily -- better just handle it transparently I'd say.